Sources/HermesAppDelegate.m

```diff
@@ -608,34 +623,52 @@ - (IBAction) updateStatusBarIcon:(id)sender {
   [statusItem setMenu:statusBarMenu];
   [statusItem setHighlightMode:YES];
 
-  statusItemImageName = @"";
   [self updateStatusBarIconImage:sender];
 }
 
 - (IBAction) updateStatusBarIconImage:(id)sender {
   if (!PREF_KEY_BOOL(STATUS_BAR_ICON))
     return;
 
-  NSString *imageName = nil;
-  if (PREF_KEY_BOOL(STATUS_BAR_ICON_BNW))
-    imageName = (playback.playing.isPlaying) ? @"Pandora-Menu-Dark-Play" : @"Pandora-Menu-Dark-Pause";
-
-  if (imageName == statusItemImageName)
-    return;
-
   NSImage *icon;
-  if (imageName == nil) {
-    icon = [[NSApp applicationIconImage] copy];
-  } else {
-    icon = [NSImage imageNamed:imageName];
+  if (PREF_KEY_BOOL(STATUS_BAR_ICON_BNW)) {
+    
+    icon = [NSImage
+            imageNamed: (playback.playing.isPlaying) ?
+            @"Pandora-Menu-Dark-Play" : @"Pandora-Menu-Dark-Pause"];
     [icon setTemplate:YES];
+    
+  } else if (PREF_KEY_BOOL(STATUS_BAR_ICON_ALBUM)) {
+    
+    if (playback.playing.isPlaying) {
+      NSData *data = [playback lastImg];
+      icon = (data) ? [[NSImage alloc] initWithData:data] :
+                      [NSImage imageNamed: @"missing-album"];
```

Comment:
Please make sure to use standard spacing (no space after colons). it's not my favored style either, but it is what Hermes does.